sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/parser/BatchParserSuite.scala

```diff
@@ -0,0 +1,187 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.parser
+
+import org.apache.spark.SparkFunSuite
+import org.apache.spark.sql.catalyst.plans.SQLHelper
+import org.apache.spark.sql.exceptions.SqlBatchLangException
+
+class BatchParserSuite extends SparkFunSuite with SQLHelper {
+  import CatalystSqlParser._
+
+  test("single select") {
+    val batch = "SELECT 1;"
+    val tree = parseBatch(batch)
+    assert(tree.collection.length == 1)
+    assert(tree.collection.head.isInstanceOf[SparkStatementWithPlan])
+    val sparkStatement = tree.collection.head.asInstanceOf[SparkStatementWithPlan]
+    assert(sparkStatement.getText(batch) == "SELECT 1;")
+  }
+
+  test("single select without ;") {
+    val batch = "SELECT 1"
+    val tree = parseBatch(batch)
+    assert(tree.collection.length == 1)
+    assert(tree.collection.head.isInstanceOf[SparkStatementWithPlan])
+    val sparkStatement = tree.collection.head.asInstanceOf[SparkStatementWithPlan]
+    assert(sparkStatement.getText(batch) == "SELECT 1")
+  }
+
+  test("multi select without ; - should fail") {
+    val batch = "SELECT 1 SELECT 1"
+    intercept[ParseException] {
+      parseBatch(batch)
+    }
+  }
+
+  test("multi select") {
+    val batch = "BEGIN SELECT 1;SELECT 2; END"
+    val tree = parseBatch(batch)
+    assert(tree.collection.length == 2)
+    assert(tree.collection.forall(_.isInstanceOf[SparkStatementWithPlan]))
+
+    batch.split(";")
+      .map(_.replace("\n", "").replace("BEGIN", "").replace("END", "").trim)
```

Comment:
Can we create helper method called `splitStatementText` or something along those lines that will do map/replace/trim thing?

Reply:
Done.